libs/langgraph/src/pregel/messages.ts

```diff
@@ -67,7 +67,13 @@ export class StreamMessagesHandler extends BaseCallbackHandler {
     // (see https://github.com/langchain-ai/langchainjs/pull/6646).
     let messageId = message.id;
     if (messageId == null || messageId === `run-${runId}`) {
-      messageId = this.stableMessageIdMap[runId] ?? messageId ?? `run-${runId}`;
+      messageId = this.stableMessageIdMap[runId] ?? messageId;
+
+      // Avoid assigning message ID for tool messages
+      // as doing so will assign same ID for all parallel tool responses.
+      // Which b/c the callback runs earlier than the reducer will cause
+      // other tools responses to be dropped.
+      if (message.getType() !== "tool") messageId ??= `run-${runId}`;
```

Comment:
nit: either `?:` or use braces - probably not a major issue with our formatting/linting setup, but sometimes people don't read closely when adding a line to a conditional block and miss that there's no curly braces.